src/Native/System.Native/pal_process.cpp

```diff
@@ -389,12 +390,20 @@ extern "C" int32_t GetPriority(PriorityWhich which, int32_t who)
 {
     // GetPriority uses errno 0 to show succes to make sure we don't have a stale value
     errno = 0;
+#if !defined(BSD) || defined(__OpenBSD__) || defined(__APPLE__)
```

Comment:
We should use CMAKE config or a pattern that would let the compiler infer the right thing here. (Also, pre-existing, but ideally, that cast to id_t would be checked with at least an assert if id_t is not wider than int32_t).  

We should not have #ifdef SPECIFIC_OS in the cpp files. (As a last, resort it can be used in config.h.in if the config test is unwieldy, see HAVE_STAT64 vs. __APPLE__)